mdp/utils/routines.py

```diff
@@ -11,6 +11,22 @@
 import random
 import itertools
 
+def _legacysignature(signature):
+    """
+    Used to replace inspect.formatargspec by inspect.signature.
+    
+    For retrocompatibility reasons, we don't use a standard Signature.
+    Instead, we use the string generated by this method.
+    Basically, from a Signature we create a string and remove the default values.
+    """
+    listsignature = str(signature)[1:-1].split(",")
+    for counter, param in enumerate(listsignature):
+        if param.count("=") > 0:
+            listsignature[counter] = param[0:param.index("=")].strip()
+        else:
+            listsignature[counter] = param.strip()
+    return ", ".join(listsignature)
+
```

Comment:
I think on top-level PEP 8 suggests two lines to separate functions. I've seen this pattern frequently in MDP so far.